src/main/webapp/jsp/instructorStudentListAjax.jsp

```diff
@@ -0,0 +1,186 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+
+<%@ page import="teammates.common.datatransfer.SectionDetailsBundle" %>
+<%@ page import="teammates.common.datatransfer.TeamDetailsBundle"%>
+<%@ page import="teammates.common.datatransfer.StudentAttributes"%>
+<%@ page import="teammates.ui.controller.InstructorStudentListAjaxPageData"%>
+<%
+    InstructorStudentListAjaxPageData data = (InstructorStudentListAjaxPageData) request.getAttribute("data");
+%>
+<%
+String COURSE_STUDENT_EDIT = "Use this to edit the details of this student. <br>"
+                           + "To edit multiple students in one go, you can use the enroll page: <br>"
+                           + "Simply enroll students using the updated data and existing data will be updated accordingly";
+String COURSE_STUDENT_DELETE = "Delete the student and the corresponding evaluations from the course";
+String COURSE_STUDENT_RECORDS = "View all student's evaluations and feedbacks";
+int courseIdx = data.courseIdx;
+int sortIdx = 2;
+if (data.courseSectionDetails.size() > 0) { %>
+    <table class="table table-responsive table-striped table-bordered margin-0">
+        <thead class="background-color-medium-gray text-color-gray font-weight-normal">
+            <tr id="resultsHeader-<%= courseIdx %>">
+                <th>Photo</th>
+                <th id="button_sortsection-<%= courseIdx %>" class="button-sort-none<%= data.hasSection ? "": " hidden" %>" onclick="toggleSort(this,<%= sortIdx %>)">
+                    Section <span class="icon-sort unsorted"></span>
+                </th>
+                <% sortIdx++; %>
+                <th id="button_sortteam-<%= courseIdx %>" class="button-sort-none" onclick="toggleSort(this,<%= sortIdx %>)">
+                    Team <span class="icon-sort unsorted"></span>
+                </th>
+                <% sortIdx++; %>
+                <th id="button_sortstudentname-<%= courseIdx %>" class="button-sort-none" onclick="toggleSort(this,<%= sortIdx %>)">
+                    Student Name <span class="icon-sort unsorted"></span>
+                </th>
+                <% sortIdx++; %>
+                <th id="button_sortemail-<%= courseIdx %>" class="button-sort-none" onclick="toggleSort(this,<%= sortIdx %>)">
+                    Email <span class="icon-sort unsorted"></span>
+                </th>
+                <% sortIdx++; %>
+                <th>Action(s)</th>
+            </tr>
+            <tr id="searchNoResults-<%= courseIdx %>" class="hidden">
+                <th class="align-center color_white bold">Cannot find students in this course</th>
+            </tr>
+        </thead>
+        <tbody>
+            <% int sectionIdx = -1;
+            int teamIdx = -1;
+            int studentIdx = -1;
+            for (SectionDetailsBundle section : data.courseSectionDetails) {
+                sectionIdx++;
+                // append to section choice, will be transported later
```

Comment:
I couldn't really understand the comments here on my first look through, only understood it when I looked at the JavaScript file and the other jsp file, perhaps can make the comments here have more information?

It's easy to look at the related parts during git diff because the affected files are put side by side and the changes too but I don't think I'll ever know where to look for the corresponding div and JavaScript if it's not indicated >.<

Maybe "Generated here but to be appended to #sectionChoices in instructorStudentList.jsp, will be transported via JavaScript in instructorStudentListAjax.js"

will help with future maintenance.